lnbits/extensions/boltz/boltz.py

```diff
@@ -0,0 +1,430 @@
+) -> Transaction:
+
+    is_refund_tx = type(swap) == SubmarineSwap
+    if is_refund_tx:
+        check_block_height(swap.timeout_block_height)
+        privkey = ec.PrivateKey.from_wif(swap.refund_privkey)
+        onchain_address = swap.refund_address
+        preimage = b""
+        sequence = 0xFFFFFFFE
+    else:
+        privkey = ec.PrivateKey.from_wif(swap.claim_privkey)
+        preimage = unhexlify(swap.preimage)
+        onchain_address = swap.onchain_address
+        sequence = 0xFFFFFFFF
+
+    locktime = swap.timeout_block_height
+    redeem_script = unhexlify(swap.redeem_script)
+
+    fees = get_fee_estimation()
+
+    tx, txid, vout_cnt, vout_amount = mempool_lockup_tx
+
+    script_pubkey = script.address_to_scriptpubkey(onchain_address)
+
+    vin = [TransactionInput(unhexlify(txid), vout_cnt, sequence=sequence)]
+    vout = [TransactionOutput(vout_amount - fees, script_pubkey)]
+    tx = Transaction(vin=vin, vout=vout)
+
+    if is_refund_tx:
+        tx.locktime = locktime
+
+    # TODO: 2 rounds for fee calculation, look at vbytes after signing and do another TX
+    s = script.Script(data=redeem_script)
+    for i, inp in enumerate(vin):
+        if is_refund_tx:
+            rs = bytes([34]) + bytes([0]) + bytes([32]) + sha256(redeem_script).digest()
+            tx.vin[i].script_sig = script.Script(data=rs)
+        h = tx.sighash_segwit(i, s, vout_amount)
+        sig = privkey.sign(h).serialize() + bytes([SIGHASH.ALL])
+        witness_items = [sig, preimage, redeem_script]
+        tx.vin[i].witness = script.Witness(items=witness_items)
+
+    return tx
+
+
+def get_swap_status(swap: Union[SubmarineSwap, ReverseSubmarineSwap]) -> SwapStatus:
+
+    swap_status = SwapStatus(
+        wallet=swap.wallet,
+        swap_id=swap.id,
+    )
+
+    try:
+        boltz_request = get_boltz_status(swap.boltz_id)
+        swap_status.boltz = boltz_request["status"]
+    except httpx.HTTPStatusError as exc:
+        json = exc.response.json()
+        swap_status.boltz = json["error"]
+        if "could not find" in swap_status.boltz:
+            swap_status.exists = False
+
+    if type(swap) == SubmarineSwap:
+        swap_status.reverse = False
+        swap_status.address = swap.address
+    else:
+        swap_status.reverse = True
+        swap_status.address = swap.lockup_address
+
+    swap_status.block_height = get_mempool_blockheight()
+    swap_status.timeout_block_height = (
+        f"{str(swap.timeout_block_height)} -> current: {str(swap_status.block_height)}"
+    )
+
+    if swap_status.block_height >= swap.timeout_block_height:
+        swap_status.hit_timeout = True
+
+    mempool_tx = get_mempool_tx(swap_status.address)
+    swap_status.lockup = mempool_tx
+    if mempool_tx == None:
+        swap_status.has_lockup = False
+        swap_status.confirmed = False
+        swap_status.mempool = "transaction.unknown"
+        swap_status.message = "lockup tx not in mempool"
```

Comment:
Keep in mind that people could still send lockup transactions for normal swaps at some random point in time after it was created (users sometimes do weird stuff)